onward/app/weather/WeatherApi.scala

```diff
@@ -54,9 +56,15 @@ class WeatherApi(wsClient: WSClient, context: ApplicationContext, actorSystem: A
   }
 
   private def getJsonWithRetry(url: String): Future[JsValue] = {
+    val weatherLogsMarkerContext: MarkerContext = MarkerContext(append("weatherRequestPath", url))
     WeatherApi.retryWeatherRequest(() => getJsonRequest(url), requestRetryDelay, actorSystem.scheduler, requestRetryMax).recover {
```

Comment:
Would this be clearer if you don't use recover? E.g.

    val resp = WeatherApi.retryWeatherRequest(() => getJsonRequest(url), requestRetryDelay, actorSystem.scheduler, requestRetryMax)

    resp.failed.foreach {
      case NonFatal(error: TimeoutException) =>
        log.warn(
          s"Request to weather api ($url) timed out (this is expected, especially at 0 and 30 mins past the hour due to" +
            s" a problem with accuweather).", error)(weatherLogsMarkerContext)
      case NonFatal(error: Throwable) =>
        log.error("Weather API request failed", error)(weatherLogsMarkerContext)
    }
    
    resp

I.e. `recover` is not idiomatically used for side-effects whereas `foreach` is I think.